util_test.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"github.com/wangii/emoji"
+	"strings"
+	"testing"
+)
+
+const unsafeCharacters = "$$$$&&&"
+
+func TestIrcSafeStringSimple(t *testing.T) {
+	// a basic emoji string
+	simpleEmojiStr := emoji.EmojiTagToUnicode(":ok_hand:" + unsafeCharacters)
+
+	// we invoke the test subject a few times in a
+	// row to test the map[string]int side effects
+	simpleSafeStr1 := IrcSafeString(simpleEmojiStr)
+	simpleSafeStr2 := IrcSafeString(simpleEmojiStr)
+	simpleSafeStr3 := IrcSafeString(simpleEmojiStr)
+
+	// ensure none of the strings match
+	if simpleSafeStr1 == simpleSafeStr2 || simpleSafeStr1 == simpleSafeStr3 || simpleSafeStr2 == simpleSafeStr3 {
```

Comment:
Also these test lines can be split nicely:

```suggestion
	if simpleSafeStr1 == simpleSafeStr2 ||
	simpleSafeStr1 == simpleSafeStr3 ||
	simpleSafeStr2 == simpleSafeStr3 {
```